### PR TITLE
Feature/FTT-891 Logger Facade

### PR DESCRIPTION
**Pull request by chrisdamba**
This PR includes tests for the logging facade using winston

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  "description": "",
+  "main": "dist/index.js",
+  "scripts": {
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "clean": "rimraf dist",
+    "build": "npm run clean && tsc"
+  },
+  "author": "",
+  "license": "ISC",
```
Comment (pjblakey) [inline package.json]: Author 'DVSA'? Also licence should be MIT i think

File: package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      "ts",
+      "tsx",
+      "js"
+    ],
+    "roots": [
+      "<rootDir>/src"
+    ],
+    "testRegex": "(/__tests__/.*|(\\.|/)(spec))\\.tsx?$"
+  }
+}
```
Comment (pjblakey) [inline package.json]: Can add a "repository" field to link the npm package to this repo -
``` "repository": "github:dvsa/azure-logger" ```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "@dvsa/azure-logger",
+  "version": "0.0.1",
+  "description": "Shareable Logging Fascade, implemented with Winston",
```
Comment (pjblakey) [inline package.json]: Typo - should be 'facade'

File: src/logger.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        }),
+        winston.format.errors({ stack: true }),
+        winston.format.splat(),
+        winston.format.json(),
+      ),
+      transports,
+    });
+  }
+
+  critical(message: string, ...optionalParams: object[]): void {
```
Comment (pjblakey) [inline src/logger.ts]: Do we know that optionalParams will always be an array of objects?

File: package.json (hunk 3 above)
Comment (chrisdamba) [reply package.json]: Fixed

File: src/logger.ts (hunk 4 above)
Comment (chrisdamba) [reply src/logger.ts]: `any[]`?

**Pull request merged**